### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by jesseklein406**

### CODE REVIEW

File: imagersite/imagersite/tests.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        link_end = mail.outbox[0].body.split('days:')[1].split()[0][18:]
+        link = '%s%s' % (self.live_server_url, link_end)
+        self.selenium.get(link)
+        self.assertEqual(
+            self.selenium.current_url,
+            '%s%s' % (self.live_server_url, '/accounts/activate/complete/')
+        )
+        self.login_helper('joseph', '123')
+        user_name = self.selenium.find_element_by_id("user-name")
+        self.assertEqual('joseph', user_name.text)
```
Comment (cewing) [inline imagersite/imagersite/tests.py:178]: yay!  Selenium tests!

File: imagersite/imagersite/views.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class IndexView(TemplateView):
+    template_name = 'index.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(IndexView, self).get_context_data(**kwargs)
+        default_photo = 'static/images/django_1024x768.png'
+        random = Photo.objects.filter(
+            published='public').order_by('?').first()
+        context['photo'] = random.photo.url if random else default_photo
+        return context
```
Comment (cewing) [inline imagersite/imagersite/views.py:15]: good.

**Comment by cewing** [comment]
nice work so far.  keep it up.

**Comment by joelstanner** [comment]
This pull request seems to have a bunch of extra stuff added from the previous assignment(s). Try and keep everything clean and only include commits that are part of the current assignment.

**Comment by jonathanstallings** [comment]
Sure thing.

File: imagersite/imager_images/templatetags/images_filters.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        Q(user=user) | Q(published='shared') | Q(published='public')
+    ).all()
+
+
+@register.filter
+def album_is_authorized(album, user):
+    # return album in Album.objects.filter(
+    #     Q(user=user) | Q(published='shared') | Q(published='public')
+    # ).all()
+    return album.published in ['shared', 'public'] or album.user.id == user.id
```
Comment (joelstanner) [inline imagersite/imager_images/templatetags/images_filters.py:32]: Nice use of custom filters. Try to keep the code clean by removing old code that is commented out when pulling to master. If you really need to get it back later, you can just refer to past commits where it was present.

File: imagersite/imager_images/tests.py [hunk 4]
```diff
@@ -128,3 +131,46 @@ def test_album_deletion_on_user_deletion(self):
...
+        for i in range(10):
+            pic = PhotoFactory(user=self.user1)
+            self.album1.photos.add(pic)
+            pic.save()
+        # self.album1.cover = pic
+        self.album1.save()
+        pic.save()
+
+    def login_helper(self, username, password):
+        self.selenium.get('%s%s' % (self.live_server_url, '/accounts/login/'))
```
Comment (joelstanner) [inline imagersite/imager_images/tests.py:163]: Selenium tests - nice.

**Pull request merged**